app/javascript/dashboard/components/widgets/conversation/helpers/SLAHelper.js

```diff
@@ -0,0 +1,135 @@
+const calculateThreshold = (timeOffset, threshold) => {
+  // Calculate the time left for the SLA to breach or the time since the SLA has breached
+  if (threshold === null) return null;
+  const currentTime = Math.floor(Date.now() / 1000);
+  return timeOffset + threshold - currentTime;
+};
+
+const findMostUrgentSLAStatus = SLAStatuses => {
+  // Sort the SLAs based on the threshold and return the most urgent SLA
+  SLAStatuses.sort((a, b) => Math.abs(a.threshold) - Math.abs(b.threshold));
+  return SLAStatuses[0];
+};
+
+const convertToMinutes = time => Math.floor(time / 60);
+const convertToHoursAndMinutes = minutes => [
+  Math.floor(minutes / 60),
+  minutes % 60,
+];
+const convertToDaysAndHours = hours => [Math.floor(hours / 24), hours % 24];
+const convertToMonthsAndDays = days => [Math.floor(days / 30), days % 30];
+const convertToYearsAndMonths = months => [
+  Math.floor(months / 12),
+  months % 12,
+];
```

Comment:
`date-fns` has functions similar to this, we can use those

Reply:
I have updated the `formatSLATime` method with the code that you shared while we were on the call.